### PR TITLE
Add yelp review dataset

### PR DESCRIPTION
**Pull request by dipam7**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

**Comment by dipam7** [comment]
When I run

```
mteb -m sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2 -t YelpReviewFullClassification
```

I get the following

```
INFO:mteb.cmd:Running with parameters: Namespace(available_tasks=False, batch_size=32, corpus_chunk_size=None, device=None, eval_splits=None, k=None, model='sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2', n_experiments=None, output_folder=None, samples_per_label=None, seed=42, task_categories=None, task_langs=None, task_types=None, tasks=['YelpReviewFullClassification'], verbosity=2)
WARNING:mteb.evaluation.MTEB:WARNING: Unknown tasks: YelpReviewFullClassification. Known tasks: ...
```

which is weird to me cz make install installs the repo in editable mode. What am I doing wrong?

**Comment by dipam7** [comment]
Also where can I find the format expected by mteb? From this [readme](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)

```
Often the dataset from HuggingFace is not in the format expected by MTEB. To resolve this you can either change the format on Hugging Face or add a dataset_transform method to your dataset to transform it into the right format on the fly.
```




 For now I just checked the `ag_news` dataset and it had the same column names as the `yelp_review` dataset

**Comment by dipam7** [comment]
Also can someone guide me on the `checking performance` part of the checklist? how do I interpret the scores? the accuracy scores are 0.37 and 0.52

File: mteb/tasks/Classification/eng/YelpReviewFullClassification.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        bibtex_citation=None,
+        n_samples={"test": 50000},
+        avg_character_length=None,
+    )
+
+    @property
+    def metadata_dict(self) -> dict[str, str]:
+        metadata_dict = dict(self.metadata)
+        metadata_dict["n_experiments"] = 10
+        metadata_dict["samples_per_label"] = 32
```
Comment (dipam7) [inline mteb/tasks/Classification/eng/YelpReviewFullClassification.py:43]: if I increase this the scores go up

File: mteb/tasks/Classification/eng/YelpReviewFullClassification.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        date=None,
+        form=["written"],
+        domains=["Reviews"],
+        task_subtypes=None,
+        license=None,
+        socioeconomic_status=None,
+        annotations_creators=None,
+        dialect=None,
+        text_creation=None,
+        bibtex_citation=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/YelpReviewFullClassification.py:34]: Try to fill these out roughly according to the paper and `mteb/abstasks/TaskMetadata.py`.
e.g. socioeconomic_status = "mixed", dialect=[], and use the citation provided by the paper.

File: mteb/tasks/Classification/eng/YelpReviewFullClassification.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        n_samples={"test": 50000},
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/YelpReviewFullClassification.py:35]: Can set this to 2048 with a variable.

File: mteb/tasks/Classification/eng/YelpReviewFullClassification.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        metadata_dict = dict(self.metadata)
+        metadata_dict["n_experiments"] = 10
+        metadata_dict["samples_per_label"] = 32
+        return metadata_dict
+
+    def dataset_transform(self):
+        total_rows = 2048
+        eval_splits = self.metadata.eval_splits
+
+        ds = {"train": self.dataset["train"]}
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/YelpReviewFullClassification.py:50]: Not sure why train set is used here. Could you please help clarify?

File: mteb/tasks/Classification/eng/YelpReviewFullClassification.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    )
+
+    @property
+    def metadata_dict(self) -> dict[str, str]:
+        metadata_dict = dict(self.metadata)
+        metadata_dict["n_experiments"] = 10
+        metadata_dict["samples_per_label"] = 32
+        return metadata_dict
+
+    def dataset_transform(self):
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/YelpReviewFullClassification.py:46]: I wonder if we can use something like this here:
```
self.dataset = self.dataset["test"].train_test_split(
            test_size=2048, seed=42, stratify_by_column="label"
   ) 
```

**Review by isaac-chung — CHANGES_REQUESTED** [review]
Great start! A few comments.
Performance-wise it looks good to me.

File: mteb/tasks/Classification/eng/YelpReviewFullClassification.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        dataset={
+            "path": "yelp_review_full",
+            "revision": "c1f9ee939b7d05667af864ee1cb066393154bf85",
+        },
+        type="Classification",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs=["eng-Latn"],
+        main_score="accuracy",
+        date=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/YelpReviewFullClassification.py:25]: Could add a rough estimate for the dates: "(start yyyy-mm-dd, end yyyy-mm-dd)".

File: mteb/tasks/Classification/eng/YelpReviewFullClassification.py (hunk 1 above)
Comment (isaac-chung) [reply mteb/tasks/Classification/eng/YelpReviewFullClassification.py:43]: That would be expected, as there are more examples for each class to be learned from.

File: mteb/tasks/Classification/eng/YelpReviewFullClassification.py (hunk 3 above)
Comment (dipam7) [reply mteb/tasks/Classification/eng/YelpReviewFullClassification.py:35]: I tried setting it to 2048 but it still uses 50000 for eval. I think this is just an info param

File: mteb/tasks/Classification/eng/YelpReviewFullClassification.py (hunk 4 above)
Comment (dipam7) [reply mteb/tasks/Classification/eng/YelpReviewFullClassification.py:50]: cz I can't throw away the train set, the experiment errors out without it

File: mteb/tasks/Classification/eng/YelpReviewFullClassification.py (hunk 2 above)
Comment (dipam7) [reply mteb/tasks/Classification/eng/YelpReviewFullClassification.py:34]: I tried adding the ones I could by looking at the dataset page and paper

File: mteb/tasks/Classification/eng/YelpReviewFullClassification.py (hunk 5 above)
Comment (dipam7) [reply mteb/tasks/Classification/eng/YelpReviewFullClassification.py:46]: Great suggestion!

File: mteb/tasks/Classification/eng/YelpReviewFullClassification.py (hunk 6 above)
Comment (dipam7) [reply mteb/tasks/Classification/eng/YelpReviewFullClassification.py:25]: Done

File: mteb/tasks/Classification/eng/YelpReviewFullClassification.py (hunk 3 above)
Comment (isaac-chung) [reply mteb/tasks/Classification/eng/YelpReviewFullClassification.py:44]: This is the metadata section, so it does not affect how many samples get used for eval from here.

**Review by isaac-chung — APPROVED** [review]
Thanks for your contributions!
